designs/v1beta1-full-changelist.md

```diff
@@ -0,0 +1,200 @@
+
+#### Remove `nodePool.spec.provider`
+
+We’ve recommended that customers leverage spec.providerRef in favor of spec.provider since Q2 2022. Documentation for this feature has been removed since Q3 2022. We will take the opportunity to remove the feature entirely to minimize code bugs/complexity and user confusion.
+
+### `NodeClass` Changes
+
+#### Update `nodeClass.spec.amiSelector`
+
+The current `amiSelector` has two primary limitations that restrict user’s ability to specify the AMIs that they want Karpenter to use:
+
+1. Users can only specify “ANDed” together requirements, meaning that if a user has an orthogonal set of tags that they want to match their images to, they have to specify them by `aws::ids` directly, since there is no way with the current tag-selection logic to specify those values
+2. Users want more flexibility to do things like specify a name/owner combination for images. Users have generally been asking Karpenter to more closely adhere to the EC2 APIs in our amiSelector design so that users can use more built-in filtering for AMIs, instead of having to use custom tagging to achieve the same outcome
+    1. To support *some* of these use-cases, Karpenter has begun effectively creating “system-tags” i.e. (`aws::ids`, `aws::owners`, `aws::name`). These are special-cased version of the standard user custom-tags that allow users to achieve the scenarios described in #2; however, they are not easily discoverable or understood and if we are beginning to support special-cases like this, it makes sense that we should begin to structure these fields.
+
+```
+amiSelectorTerms:
+- name: foo
+  id: abc-123
+  owner: amazon
+  ssm: "/al2/alias"
+  tags: 
+    key: value
+# Selectors are ORed
+- name: foo
+  id: abc-123
+  owner: self
+  ssm: "/al2/alias"
+  tags: 
+    key: value
+```
+
+#### Update `nodeClass.spec.subnetSelector`
+
+`subnetSelectorTerms` should have a similar parity to the `amiSelectorTerms` in its design to improve the ease-of-use for users. As a result, we should design the `subnetSelectorTerms` in the same spirit as the `amiSelectorTerms` such that you can also specify multiple selectors through `tags` and `ids` that can be ORed together to produce the ultimate set of items that you want to use.
+
+```
+subnetSelectorTerms:
+- id: abc-123
+  tags:
+    key: value
+# Selectors are ORed
+- id: abc-123
+  tags:
+    key: value
+```
+
+#### Update `nodeClass.spec.securityGroupSelector`
+
+The same logic for `subnetSelectorTerms` applies to `securityGroupSelectorTerms`. We should have a similar parity to the `amiSelectorTerms` to improve the ease-of-use around this selector.
+
+```
+securityGroupSelectorTerms:
+- id: abc-123
+  name: default-secuirty-group # Not the same as the name tag
+  tags:
+    key: value
+# Selectors are ORed
```

Comment:
I reread how the actual [`DescribeSecurityGroups`](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSecurityGroups.html) API works. You can specify multiple _values_ (eg, ask for two different tag values for the same key, and accept either) but the “key” part requirements are ANDed together.

Reply:
Yep. Maybe what's confusing is how I worded the comment. What I _meant_ was that the selector terms are ORed but the actual selector itself is ANDed.